shadows/framework/src/main/java/org/robolectric/shadows/ShadowPowerManager.java

```diff
@@ -537,6 +537,16 @@ protected synchronized boolean isHeld() {
       return !timeoutTimestampList.isEmpty();
     }
 
+    @Override
+    @Implementation
+    public String toString() {
```

Comment:
Shadowing toString is usually not the right approach. It is not fixing the root cause.

If you are seeing an NPE, it means something is not initialized.

Reply:
The root cause is that Shadow.newInstanceOf is used here:
https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowPowerManager.java#L94

It means that nothing in WakeLock actually gets initialized.

A better short-term fix is to ensure that `mToken` is initialized here, at least.

Reply:
If possible, I would try to invoke the real Android code for `newWakeLock`:

https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/os/PowerManager.java;l=1352-1357

You could either try removing the shadow for newWakeLock, or calling the original code first using a `@Direct` reflector.

I can take a look at this more closely tomorrow and make more concrete suggestions.

Reply:
I just followed the existing implementation:
https://github.com/robolectric/robolectric/blob/8dfa3194cd0cdf4967fe6492fe341a59c63fde26/shadows/framework/src/main/java/org/robolectric/shadows/ShadowAccessibilityWindowInfo.java#L369
https://github.com/robolectric/robolectric/blob/8dfa3194cd0cdf4967fe6492fe341a59c63fde26/shadows/framework/src/main/java/org/robolectric/shadows/ShadowAccessibilityNodeInfo.java#L688

I've also noticed other shadows that implement `toString()` without `@Implementation` annotation:
https://github.com/robolectric/robolectric/blob/8dfa3194cd0cdf4967fe6492fe341a59c63fde26/shadows/framework/src/main/java/org/robolectric/shadows/ShadowAccessibilityNodeInfo.java#L688

This PR can also be modified to remove `@Implementation` from `toString()` and it will work.